src/main/java/org/opentripplanner/graph_builder/module/osm/ParkingProcessor.java

```diff
@@ -275,6 +277,41 @@ private VehicleParking buildParkAndRideAreasForGroup(AreaGroup group, boolean is
     return vehicleParking;
   }
 
+  /**
+   * Creates an artificial entrance to a parking facility's centroid.
+   * <p>
+   * This is useful if the facility is not linked to the street network in OSM. Without this method
+   * it would not be usable by the routing algorithm as it's unreachable.
+   */
+  private List<VehicleParking.VehicleParkingEntranceCreator> createArtificialEntrances(
+    AreaGroup area,
+    I18NString vehicleParkingName,
+    OSMWithTags entity,
+    boolean isCarPark
+  ) {
+    var centroid = area.outermostRings.get(0).jtsPolygon.getCentroid();
```

Comment:
This might produce a result outside the parking place

Reply:
Should I union all the polygons instead and get the centroid from that?